retrievalmarket/impl/provider.go

```diff
@@ -51,6 +54,7 @@ type Provider struct {
 	dealDecider          DealDecider
 	askStore             retrievalmarket.AskStore
 	disableNewDeals      bool
+	dealPricingFunc      DealPricingFunc
```

Comment:
"Deal" is too general a term. Let's remove the ambiguity here.

```suggestion
	retrievalPricingFunc      RetrievalPricingFunc
```